cns/cniconflist/generator_linux.go

```diff
@@ -0,0 +1,75 @@
+package cniconflist
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/Azure/azure-container-networking/cni"
+	cninet "github.com/Azure/azure-container-networking/cni/network"
+	"github.com/Azure/azure-container-networking/cni/util"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/network"
+)
+
+const (
+	portmapType            = "portmap"
+	portMappingsCapability = "portMappings"
+)
+
+var errUnsupportedScenario = errors.New("unsupported scenario")
+
+func NewGenerator(w io.WriteCloser, scenario configuration.CNIConflistScenario) (*Generator, error) {
+	g := Generator{writer: w}
+	switch scenario {
+	case configuration.CNIConflistScenarioV4Overlay:
+		g.scenario = scenario
+	default:
+		return nil, fmt.Errorf("can't build generator: %w", errUnsupportedScenario)
+	}
+
+	return &g, nil
+}
+
+func (g *Generator) Generate() error {
+	var azureVnetConfig cni.NetworkConfig
+	if g.scenario == configuration.CNIConflistScenarioV4Overlay {
```

Comment:
I thought about that but kind of expected the scenarios to share a lot of code, and basically just change a couple of lines like `executionMode` based on the scenario, so I thought sharing the code in a single generator made more sense. But it is a little bit harder to discern what actually gets generated since you kind of have to trace the code mentally to figure it out. I kind of accepted that tradeoff because we have the testdata/fixtures to see exactly what we expect to get generated, but I may refactor a bit to try our your suggestion and see which one I like better.